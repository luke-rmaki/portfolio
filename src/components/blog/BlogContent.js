import styled from 'styled-components';
import mediaQueries from '../../utils/mediaQueries';

const BlogContent = styled.div`
  * {
    padding: 0 25px;

    @media screen and (min-width: 800px) {
      padding: 0 100px;
    }
  }
  h2 {
    font-family: 'Monserrat', sans-serif;
    font-weight: normal;
    font-style: italic;
    font-size: 30px;
    color: var(--main);

    ${mediaQueries.typography} {
      font-size: 48px;
    }
  }

  p {
    font-family: 'Monserrat', sans-serif;
    font-size: 18px;
    color: var(--white);
    padding: 10px 25px;

    ${mediaQueries.typography} {
      padding: 30px 100px;
      font-size: 20px;
    }
  }

  /* Image and caption container */
  figure {
    padding: 0;

    img {
      padding: 0;
    }

    figcaption {
      text-align: center;
      color: var(--main-shade);
      font-family: 'Monserrat', sans-serif;
    }
  }
`;

export default BlogContent;
