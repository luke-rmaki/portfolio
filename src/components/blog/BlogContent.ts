import styled from 'styled-components';

const BlogContent = styled.article`
  padding: 50px 0 150px 0;

  h2,
  h3,
  h4 {
    font-family: var(--font);
    padding: 0 25px;
  }

  h2 {
    font-weight: normal;
    font-style: italic;
    font-size: 1.8rem;
    color: var(--accent);
    margin: 50px 0 20px 0;
  }

  h3 {
    color: var(--purple-light-2);
    font-size: 1.3rem;
    margin: 50px 0 0 0;
  }

  h4 {
    color: var(--white);
    font-size: 1.1rem;
    margin-top: 25px;
    font-style: italic;
  }

  p {
    font-family: var(--font);
    font-size: 1rem;
    color: var(--white);
    padding: 10px 25px;
    line-height: 1.8rem;

    strong {
      padding: 0;
      background: var(--accent);
      color: var(--black);
    }

    em {
      padding: 0;
    }
  }

  hr {
    margin-top: 40px;
  }

  deckgo-highlight-code {
    margin: 40px 0;
  }

  blockquote {
    background: var(--purple);
    margin: 10px 0;
  }

  /* Image and caption container */
  figure {
    padding: 0;

    img {
      padding: 0;
    }

    figcaption {
      text-align: center;
      color: var(--purple-light-1);
      font-family: var(--font);
      font-size: 0.8rem;
    }
  }

  ul {
    padding: 10px 25px;
    li {
      list-style-type: none;
      color: var(--white);
      &::before {
        content: '=> ';
      }
    }
  }
  /*  Desktop styles */

  @media (min-width: 1000px) {
    h2,
    h3,
    h4,
    p,
    blockquote {
      padding: 0 100px;
    }

    blockquote {
      margin: 40px 0;
      p {
        padding: 30px 0;
      }
    }

    deckgo-highlight-code {
      width: 80vw;
      margin: 50px auto;
    }
  }
`;

export default BlogContent;
