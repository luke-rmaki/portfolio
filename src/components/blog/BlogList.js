import React from 'react';
import styled from 'styled-components';
import Img from 'gatsby-image';
import { Link } from 'gatsby';

import { ContentSpring } from '../springs/about';

const BlogListWrapper = styled.div`
  display: flex;
  flex-direction: column;

  @media screen and (min-width: 600px) {
    flex-direction: row;
    flex-wrap: wrap;

    > * {
      flex: 1 0 50px;
      padding: 10px;
    }
  }
`;

const BlogListing = styled.div`
  position: relative;
  margin: 20px 0;

  div#excerpt {
    background-color: rgba(211, 212, 244, 0.5);
    position: absolute;
    bottom: 0;
    height: 100px;
    color: var(--black);
    margin: auto;
    padding: 10px;
  }
`;

const BlogList = list => {
  return (
    <BlogListWrapper>
      {list.list.map(edge => (
        <ContentSpring key={edge.node.frontmatter.path}>
          <BlogListing>
            <Link to={edge.node.frontmatter.path}>
              <Img fluid={edge.node.frontmatter.image.childImageSharp.fluid} />
              <div id="excerpt">{edge.node.excerpt}</div>
            </Link>
          </BlogListing>
        </ContentSpring>
      ))}
    </BlogListWrapper>
  );
};

export default BlogList;
