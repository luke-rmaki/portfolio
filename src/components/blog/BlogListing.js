import styled from 'styled-components';

const BlogListing = styled.div`
  position: relative;
  margin: 20px 0;

  div#excerpt {
    overflow: hidden;
    h2 {
      font-size: 2rem;
      padding-bottom: 20px;
    }
    p {
      font-size: 1rem;
    }
    background-color: rgba(211, 212, 244, 0.5);
    position: absolute;
    bottom: 0;
    height: 150px;
    color: var(--black);
    margin: auto;
    padding: 10px;
  }
`;

export default BlogListing;
