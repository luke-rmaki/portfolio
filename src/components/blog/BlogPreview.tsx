import React from 'react';
import styled from 'styled-components';
import { GatsbyImage, getImage } from 'gatsby-plugin-image';
import { Link } from 'gatsby';
import { Paragraph, SubHeading } from '../Typography';
import { Frontmatter } from '../../utils/types';

type Props = {
  edge: {
    node: {
      excerpt: string;
      frontmatter: Frontmatter;
    };
  };
};
export const BlogPreview = (props: Props) => {
  const { edge } = props;
  return (
    <StyledBlogPreview>
      <Link to={edge.node.frontmatter.path}>
        <section className="excerpt">
          <SubHeading>{edge.node.frontmatter.title}</SubHeading>
          <div className="sub">
            <svg width="100%" height="30px">
              <line x1="50%" x2="50%" y1="0" y2="90%" />
              <line x1="50%" x2="95%" y1="90%" y2="90%" />
            </svg>
            <span>{edge.node.frontmatter?.date}</span>
          </div>
          <Paragraph>{edge.node.excerpt}</Paragraph>
        </section>
        <GatsbyImage
          alt="Blog post header image"
          image={getImage(edge.node.frontmatter.image)}
        />
      </Link>
    </StyledBlogPreview>
  );
};

const StyledBlogPreview = styled.article`
  position: relative;
  margin: 50px 0;
  transition: 0.2s all ease-in;

  a {
    text-decoration: none;
    display: flex;
    flex-direction: column;
  }

  /* Desktop layout */
  @media (min-width: 1000px) {
    padding: 0 100px;

    a {
      flex-direction: row;
      gap: 50px;

      .excerpt {
        flex: 0 1 50%;
      }
    }
  }

  h2 {
    text-decoration: underline;
    font-style: normal;
    color: var(--white);
    padding: 0 20px;
  }

  .sub {
    display: flex;
    align-items: flex-end;
    height: 30px;
    margin-bottom: 50px;

    span {
      color: var(--white);
      font-family: var(--font);
    }

    svg {
      line {
        stroke: var(--accent);
        stroke-width: 1;
      }
    }
  }

  p {
    padding: 0 20px;
  }

  section {
    margin-bottom: 25px;
  }

  &:hover {
    transform: scale(1.05);
    box-shadow: 4px 0px 6px 4px var(--black-dark-1);
    h2,
    p,
    span {
      color: var(--accent);
    }

    svg > line {
      stroke: var(--white);
    }
  }
`;
