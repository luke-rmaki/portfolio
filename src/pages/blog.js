import React from 'react';
import { graphql } from 'gatsby';

// Component imports
import { BlogList } from '../components/blog';
import Menu from '../components/global/Menu';
import { Layout, Spacer, Flex } from '../components/global/layout';
import { MainHeading } from '../components/global/typography';
import { HeadingSpring } from '../components/global/springs';

const Blog = (
  {
    data: {
      allMarkdownRemark: { edges },
    },
  },
  props
) => {
  return (
    <Layout>
      <Menu path={props.path} />
      <Spacer heightMobile={150} />
      <HeadingSpring>
        <MainHeading style={{ paddingLeft: '25px' }}>Blog</MainHeading>
      </HeadingSpring>
      <BlogList list={edges} />
    </Layout>
  );
};

export default Blog;

export const pageQuery = graphql`
  query {
    allMarkdownRemark(
      filter: { frontmatter: { type: { ne: "page" }, path: { ne: null } } }
      sort: { order: DESC, fields: [frontmatter___date] }
    ) {
      edges {
        node {
          excerpt(pruneLength: 200)
          frontmatter {
            date(formatString: "DD MMMM, YYYY")
            path
            title
            image {
              childImageSharp {
                fluid(maxWidth: 800) {
                  ...GatsbyImageSharpFluid
                }
              }
            }
          }
        }
      }
    }
  }
`;
