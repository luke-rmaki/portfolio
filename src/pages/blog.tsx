import React from 'react';
import { graphql } from 'gatsby';

// Component imports
import { BlogList } from '../components/blog';
import { Layout, Menu } from '../components/global';
import { MainHeading } from '../components';

const Blog = (
  {
    data: {
      allMdx: { edges },
    },
  },
  props
) => {
  const { path }: { path: string } = props;
  return (
    <Layout>
      <Menu path={path} />
      <MainHeading>Blog</MainHeading>
      <BlogList list={edges} />
    </Layout>
  );
};

export default Blog;

export const pageQuery = graphql`
  query {
    allMdx(
      filter: { frontmatter: { title: { ne: "about" }, path: { ne: null } } }
      sort: { order: DESC, fields: [frontmatter___date] }
    ) {
      edges {
        node {
          excerpt(pruneLength: 200)
          frontmatter {
            date(formatString: "DD MMMM, YYYY")
            path
            title
            image {
              childImageSharp {
                fluid(maxWidth: 800) {
                  ...GatsbyImageSharpFluid
                }
              }
            }
          }
        }
      }
    }
  }
`;
