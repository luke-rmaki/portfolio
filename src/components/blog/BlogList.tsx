import React from 'react';
import styled from 'styled-components';

import { BlogPreview } from './BlogPreview';
import { BlogOverviewData } from '../../utils/types';

type Props = {
  list: BlogOverviewData[`allMdx`][`edges`];
};
export const BlogList = (props: Props) => {
  const { list } = props;
  return (
    <BlogListWrapper>
      {list.map((edge) => (
        <BlogPreview
          key={edge.node.frontmatter.path}
          data-cy="blog-post"
          edge={edge}
        />
      ))}

      <hr />
    </BlogListWrapper>
  );
};

const BlogListWrapper = styled.div`
  display: flex;
  flex-direction: column;

  /* @media (min-width: 1000px) { */
  /* padding: 0 100px; */
  /* } */
`;
