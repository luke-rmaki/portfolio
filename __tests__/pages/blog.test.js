import React from 'react';
import renderer from 'react-test-renderer';

import blog from '../../src/pages/index';

describe('blog', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<blog />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
