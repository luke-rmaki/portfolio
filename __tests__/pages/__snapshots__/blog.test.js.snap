// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blog renders correctly 1`] = `<blog />`;
